pkg/ddevapp/ddevapp.go

```diff
@@ -888,7 +934,11 @@ func (app *DdevApp) SnapshotDatabase(snapshotName string) (string, error) {
 	}
 
 	util.Warning("Creating database snapshot %s", snapshotName)
-	stdout, stderr, err := app.Exec("db", "bash", "-c", fmt.Sprintf("mariabackup --backup --target-dir=%s --user root --password root --socket=/var/tmp/mysql.sock 2>/var/log/mariadbackup_backup_%s.log && cp /var/lib/mysql/db_mariadb_version.txt %s", containerSnapshotDir, snapshotName, containerSnapshotDir))
+	stdout, stderr, err := app.Exec(&ExecOpts{
+		Service: "db",
+		Cmd:     []string{"bash", "-c", fmt.Sprintf("mariabackup --backup --target-dir=%s --user root --password root --socket=/var/tmp/mysql.sock 2>/var/log/mariadbackup_backup_%s.log && cp /var/lib/mysql/db_mariadb_version.txt %s", containerSnapshotDir, snapshotName, containerSnapshotDir)},
```

Comment:
This ultimately comes down to personal preference, but I think there are some real benefits to using a single structured argument in cases like this where there are implicit, optional, or default values.  The preference is to avoid passing in a mushy series of strings as an argument where some are required, some aren't, and it can be difficult to read:
`app.Exec("db", "", "ls", "-al", "/usr/local/bin")`

I don't think this is always (or even usually) appropriate because it removes the ability to rely on the compiler to catch issues with arguments not matching a function signature, and it can put additional responsibility on the function to do some validation of the inputs. In this case, though, we're already in a difficult situation because Exec() is variadic and only accepts string inputs. Adding another string argument means any call where `len(cmd) > 1` will compile just fine and all your positional arguments are now off by one.